tests/integration.rs

```diff
@@ -870,6 +888,113 @@ mod integration_tests {
         Ok(())
     }
 
+    #[cfg(not(target_os = "windows"))]
+    #[test]
+    fn test_spin_plugin_install_command() -> Result<()> {
+        // Create a temporary directory for plugin source and manifests
+        let temp_dir = tempdir()?;
+        let dir = temp_dir.path();
+        let installed_plugins_dir = dir.join("tmp");
+
+        // Ensure that spin installs the plugins into the temporary directory
+        let mut env_map: HashMap<&str, &str> = HashMap::new();
+        env_map.insert(
+            "TEST_PLUGINS_DIRECTORY",
+            installed_plugins_dir.to_str().unwrap(),
+        );
+
+        let path_to_test_dir = std::env::current_dir()?;
+        // TODO: test on Windows
+        let plugin_manifest_json_template = r#"
+        {
+            "name": "example",
+            "description": "A description of the plugin.",
+            "homepage": "www.example.com",
+            "version": "0.2.0",
+            "spinCompatibility": ">=0.4",
+            "license": "MIT",
+            "packages": [
+                {
+                    "os": "linux",
+                    "arch": "amd64",
+                    "url": "file:PATH_TO_TESTS/tests/plugin/example.tar.gz",
+                    "sha256": "f7a5a8c16a94fe934007f777a1bf532ef7e42b02133e31abf7523177b220a1ce"
+                },
+                {
+                    "os": "macos",
+                    "arch": "aarch64",
+                    "url": "file:PATH_TO_TESTS/tests/plugin/example.tar.gz",
+                    "sha256": "f7a5a8c16a94fe934007f777a1bf532ef7e42b02133e31abf7523177b220a1ce"
+                },
+                {
+                    "os": "macos",
+                    "arch": "amd64",
+                    "url": "file:PATH_TO_TESTS/tests/plugin/example.tar.gz",
+                    "sha256": "f7a5a8c16a94fe934007f777a1bf532ef7e42b02133e31abf7523177b220a1ce"
+                }
+            ]
+        }"#;
+        let plugin_manifest_json = plugin_manifest_json_template
+            .replace("PATH_TO_TESTS", path_to_test_dir.to_str().unwrap());
```

Comment:
You might like `serde_json::json!` for this, which lets you write things like:

```rust
let plugin_manifest_json = serde_json::to_vec(serde_json::json!({
  // ...
  "url": format!("file:{PATH_TO_TESTS}/tests/plugin/example.tar.gz")
  // ...
})).unwrap();
```

Reply:
This was much cleaner. Thanks!